cmd/terramate/cli/cloud.go

```diff
@@ -477,30 +457,87 @@ func (c *cli) detectCloudMetadata() {
 	}
 }
 
-func getGithubCommit(ghClient *github.Client, repo string, commitName string) (*github.Commit, error) {
+func getGithubCommit(ghClient *github.Client, owner, repo, commit string) (*github.RepositoryCommit, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultGithubTimeout)
 	defer cancel()
 
-	commit, err := ghClient.Commit(ctx, repo, commitName)
+	rcommit, _, err := ghClient.Repositories.GetCommit(ctx, owner, repo, commit, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	return commit, nil
+	return rcommit, nil
 }
 
-func getGithubPR(ghClient *github.Client, repo string, commitName string) (github.Pulls, error) {
+// returns nil, nil if there was no PR associated with commit
+func getGithubPR(ghClient *github.Client, owner, repo, commit string) (*github.PullRequest, error) {
```

Comment:
I think it's better to change the return to be `(pull *github.PullRequest, found bool, err error)`.

In idiomatic Go, usually whenever `err == nil` then the value is ready to be used, so I would expect:
```go
pr, err := getGithubPR(client, owner, repo, commit)
if err != nil {
   // handle error: continue or return
}

st := pull.GetState() // pr can be safely used.
```

but for the case where the value is empty and no error occurs, then usually I return an additional boolean flag that hints if the value can be used. So it would become:

```go
pr, found, err := getGithubPR(...)
if err != nil {
   // handle err
}
if found {
   // safely to use pr
   pr.GetState()
}
```